Simplenote/Classes/SPNoteListViewController+Extensions.swift

```diff
@@ -535,9 +606,18 @@ extension SPNoteListViewController: UITableViewDelegate {
     }
 
     public func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        guard let cell = cell as? SPNoteTableViewCell else {
+            return
+        }
+
         var insets = SPNoteTableViewCell.separatorInsets
         insets.left -= cell.layoutMargins.left
+
         cell.separatorInset = insets
+
+        if indexPath.row == notesListController.numberOfObjects - 1 {
+            cell.shouldDisplayBottomSeparator = false
+        }
```

Comment:
Because of cell reuse, a given Cell might have the `shouldDisplayBottomSeparator` set to `false`, and that value is never set back to true.

Always always consider doing something like:

```
cell.shouldDisplayBottomSeparator = indexPath.row < notesListController.numberOfObjects - 1 
```

**Furthermore** the AppStore build doesn't hide the cell separator for the last row. I think you can drop this snippet?

Reply:
Asked @SylvesterWilmott if we wanted to have the last separator and he said that we shouldn't currently have it and that I could drop it if I wanted.  My sense was that with the new note bar, the two separators join weirdly and it would be better without it.

Made this change